torchao/quantization/qat/utils.py

```diff
@@ -16,6 +19,37 @@
 )
 
 
+class _Float8FakeQuantize(torch.autograd.Function):
+    """
+    Implementation of float8 fake quantize with backward STE.
+    """
+
+    @staticmethod
+    def forward(
+        ctx: torch.autograd.function.FunctionCtx,
+        x: torch.Tensor,
+        float8_dtype: torch.dtype,
+        scaling_granularity: ScalingGranularity,
+        axiswise_dim: Optional[int],
+    ):
+        original_dtype = x.dtype
+        scale = tensor_to_scale(
+            x,
+            float8_dtype,
+            scaling_granularity=scaling_granularity,
```

Comment:
IMO better to just put the granularity you want here to also remove it from the middle layer